src/layer/Layer.js

```diff
@@ -78,6 +79,27 @@ export var Layer = Evented.extend({
 		return this._map.getPane(name ? (this.options[name] || name) : this.options.pane);
 	},
 
+	// @method setPane(name : String): this
+	// The layer will be moved to another pane
+	setPane: function (name) {
+		if (!this._map) {
+			this.options.pane = name;
+			return this;
+		}
+
+		if (this instanceof L.Marker) {
+			DomUtil.remove(this._icon);
+			this.options.pane = name;
+			this.getPane().appendChild(this._icon);
+		} else {
+			this._map.getRenderer(this)._removePath(this);
+			this.options.pane = name;
```

Comment:
nit-pick:  The above line is included in every branch of the function.  Could you call it once and remove duplication? e.g.

```
setPane: function (name) {
		this.options.pane = name;

		if (!this._map) {
			return this;
		}

		if (this instanceof L.Marker) {
			DomUtil.remove(this._icon);
			this.getPane().appendChild(this._icon);
		} else {
			this._map.getRenderer(this)._removePath(this);
			this._renderer = this._map.getRenderer(this);
			this.onAdd();
		}
		return this;
	},
```

Reply:
Yes this is not looking well, but i have not found another solution as this, because `options.pane` have to be the old before `this._map.getRenderer(this)._removePath(this);` is called, else it is not removed clean.